src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer.UnitTests/VirtualProjectXmlProviderTests.cs

```diff
@@ -0,0 +1,189 @@
+        var projectProvider = await GetProjectXmlProviderAsync();
+
+        var tempDir = TempRoot.CreateDirectory();
+        var appFile = tempDir.CreateFile("app.cs");
+        await appFile.WriteAllTextAsync("""
+            public class C
+            {
+            }
+            """);
+
+        var globalJsonFile = tempDir.CreateFile("global.json");
+        await globalJsonFile.WriteAllTextAsync("""
+            {
+              "sdk": {
+                "version": "10.0.100-preview.5.25265.12"
+              }
+            }
+            """);
+
+        var contentNullable = await projectProvider.GetVirtualProjectContentAsync(appFile.Path, CancellationToken.None);
+        var content = contentNullable.Value;
+        LoggerFactory.CreateLogger<VirtualProjectXmlProviderTests>().LogTrace(content.VirtualProjectXml);
+
+        Assert.Contains("<TargetFramework>net10.0</TargetFramework>", content.VirtualProjectXml);
+        Assert.Contains("<ArtifactsPath>", content.VirtualProjectXml);
+        Assert.Empty(content.Diagnostics);
+    }
+
+    [ConditionalFact(typeof(EnableRunApiTests))]
+    public async Task GetProjectXml_BadPath_01()
+    {
+        var projectProvider = await GetProjectXmlProviderAsync();
+
+        var tempDir = TempRoot.CreateDirectory();
+
+        var globalJsonFile = tempDir.CreateFile("global.json");
+        await globalJsonFile.WriteAllTextAsync("""
+            {
+              "sdk": {
+                "version": "10.0.100-preview.5.25265.12"
+              }
+            }
+            """);
+
+        var content = await projectProvider.GetVirtualProjectContentAsync(Path.Combine(tempDir.Path, "BAD"), CancellationToken.None);
+        Assert.Null(content);
+    }
+
+    [ConditionalFact(typeof(EnableRunApiTests))]
+    public async Task GetProjectXml_BadDirective_01()
+    {
+        var projectProvider = await GetProjectXmlProviderAsync();
+
+        var tempDir = TempRoot.CreateDirectory();
+        var appFile = tempDir.CreateFile("app.cs");
+        await appFile.WriteAllTextAsync("""
+            #:package Newtonsoft.Json@13.0.3
+            #:BAD
+            Console.WriteLine("Hello, world!");
+            """);
+
+        var globalJsonFile = tempDir.CreateFile("global.json");
+        await globalJsonFile.WriteAllTextAsync("""
+            {
+              "sdk": {
+                "version": "10.0.100-preview.5.25265.12"
+              }
+            }
+            """);
+
+        var contentNullable = await projectProvider.GetVirtualProjectContentAsync(appFile.Path, CancellationToken.None);
+        var content = contentNullable.Value;
+        var diagnostic = content.Diagnostics.Single();
+        Assert.Contains("Unrecognized directive 'BAD'", diagnostic.Message);
+        Assert.Equal(appFile.Path, diagnostic.Location.Path);
+
+        // LinePositionSpan is not deserializing properly.
+        // Address when implementing editor squiggles. https://github.com/dotnet/roslyn/issues/78688
```

Comment:
Seems like we should fix that now, since otherwise we're reporting bad locations to the error list?